WalletWasabi/WabiSabi/Client/LiquidityClueProvider.cs

```diff
@@ -0,0 +1,93 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using WalletWasabi.Blockchain.Analysis;
+using WalletWasabi.Extensions;
+using WalletWasabi.Helpers;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+public class LiquidityClueProvider
+{
+	public LiquidityClueProvider()
+	{
+		LiquidityClue = null;
+		LiquidityClueLock = new object();
+	}
+
+	private Money? LiquidityClue { get; set; }
+	private object LiquidityClueLock { get; }
+
+	public void InitLiquidityClue(IEnumerable<Money> foreignOutputsValues)
+	{
+		lock (LiquidityClueLock)
+		{
+			if (TryCalculateLiquidityClue(foreignOutputsValues, out var newLiquidityClue))
+			{
+				LiquidityClue = newLiquidityClue;
+			}
+		}
+	}
+
+	public void InitLiquidityClue(Transaction lastCoinJoin, IEnumerable<TxOut> walletTxOuts) =>
+		InitLiquidityClue(GetForeignOutputsValues(lastCoinJoin, walletTxOuts));
+
+	public async Task InitLiquidityClueAsync(IWallet wallet, CancellationToken cancellationToken = default)
+	{
+		var transactions = await wallet.GetTransactionsAsync().ConfigureAwait(false);
+		if (transactions.OrderByBlockchain().LastOrDefault(x => x.IsOwnCoinjoin()) is { } lastCoinJoin)
+		{
+			InitLiquidityClue(lastCoinJoin.Transaction, lastCoinJoin.WalletOutputs.Select(x => x.TxOut));
+		}
+	}
+
+	public Money GetLiquidityClue(Money maxSuggestedAmount)
+	{
+		lock (LiquidityClueLock)
+		{
+			return Math.Min(
+				LiquidityClue ?? Constants.MaximumNumberOfBitcoinsMoney,
```

Comment:
```suggestion
				LiquidityClue ?? ProtocolConstants.MaxAmountPerAlice,
```

Reply:
Oh and CF is complaining.

Reply:
The liquidity is based on the max suggested amount which is unbounded.